api/types/client.go

```diff
@@ -181,8 +181,20 @@ type ImageBuildOptions struct {
 	Target      string
 	SessionID   string
 	Platform    string
+	Version     BuilderVersion
+	BuildID     string
 }
 
+// BuilderVersion sets the version of underlying builder to use
+type BuilderVersion string
+
+const (
+	// BuilderV1 is the first generation builder in docker daemon
+	BuilderV1 BuilderVersion = "1"
+	// BuilderBuildKit is builder based on moby/buildkit project
+	BuilderBuildKit = "2"
+)
```

Comment:
Nit: can we just use strings rather than numbers? e.g. `legacy` and `buildkit`

Reply:
@AkihiroSuda I don't like the word `legacy` (which is what it was initially). With all the respect I have for buildkit, with time everything becomes legacy and eventually you'll have to differentiate between `legacy legacy` and `legacy`. This problem is solved with versions, so I believe it's a better tradeoff this way. Happy to hear your concerns though.